bin/release.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+
+VERSION=`ruby -r ./lib/event_sourcing/version.rb -e "puts EventSourcing::VERSION"`
+
+echo "--- Building gem"
+gem build event-sourcing.gemspec
+
+echo "--- Tagging ${VERSION} on git"
+git tag $VERSION
+git push --tags
+
+echo "--- Pushing ${VERSION} to Rubygems.org"
+gem push "event-sourcing-${VERSION}.gem"
+
+rm "event-sourcing-${VERSION}.gem"
```

Comment:
As you may notice, this script doesn't seem idempotent (re-releasing a version would be possible). Each step of this release shell protects the developer of making mistakes:

- Git: creating an existent tag version will raise an error telling it the tag already exists
- Rubygems: `gem push` won't allow repushing gems

Since the two commands that interact with remote servers are well protected against user errors, I decided to not include the `set -e` bash flag, as it might be interesting to read the error feedback from each command.

Reply:
Sure, you can create a [variable](https://www.gnu.org/software/make/manual/html_node/Setting.html#Setting) at the top of your makefile:

```makefile
version := $(shell ruby -r ./lib/event_sourcing/version.rb -e "puts EventSourcing::VERSION")
```

and then use it elsewhere in the makefile:

```sh
git tag $(version)
```

Reply:
Oh! That's interesting! Will do!